### ISSUE TITLE
page.screenshot full page option broken

### ISSUE THREAD

**Issue by bernabe9** [puppeteer/puppeteer#6569]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 5.4.1
* Platform / OS version: macOS Catalina 
* URLs (if applicable):
* Node.js version: 12

**What steps will reproduce the problem?**

```javascript
// script.js

const puppeteer = require('puppeteer')

const main = async () => {
  const browser = await puppeteer.launch()
  const page = await browser.newPage()

  await page.goto('https://github.com/puppeteer/puppeteer/releases')

  await page.screenshot({ path: 'screenshot.png', fullPage: true })
  await browser.close()
}

main()
```

1. Run the script `node script.js`
2. See the result in `screenshot.png`

**What is the expected result?**
The full-page screenshot should be displayed properly

**What happens instead?**
This is the result (overlapped and duplicated screenshot)
<kbd>![screenshot](https://user-images.githubusercontent.com/18464392/97818915-121bba80-1c84-11eb-8228-1c58b0bca1b2.png)</kbd>

**Comment by dcosta-ptc** [comment]
Hi,

Same here, was updating puppeteer to latest version, and noticed was having a error in our visual tests regarding a page with 15k heigh page exactly like mentioned above. Page starts repeating after some height, in my case is around 8500px height.

Sample test created to test:
```js
const puppeteer = require('puppeteer'); 
const url = "http://localhost/samplebigpage/";
async function run() {     
   let browser = await puppeteer.launch({ headless: true });
   let page = await browser.newPage();
   await page.goto(url, { waitUntil: "networkidle0", timeout: 60000 });
   await page.setViewport({ width: 1280, height: 1024 });
   await page.screenshot({
    path: "./screenshot.png",
    type: "png",
    fullPage: true,
  });
  await page.close();
  await browser.close();
} 
run();
```

Running this on latest versions of puppeteer 2, 3 and 4 gives a proper full screenshot. With latest 5 happens what is described.

**Update:**
Tested the same but with headless: false and it worked properly. Maybe it's a headless mode only issue.

**Comment by bernabe9** [comment]
@dcosta-ptc good catch! Works ok when `headless` flag is `false`

**Comment by dcosta-ptc** [comment]
Appears to have been fixed on v5.5.0.
Can you confirm @bernabe9?

**Comment by bernabe9** [comment]
Looks fixed on v5.5.0. Closing this issue

**Comment by bartolootrit** [comment]
Puppeteer 5.5.0, Ubuntu 20.04. The bug exists with `headless: false`:

```
const puppeteer = require('puppeteer'); 
    const url = "https://github.com/puppeteer/puppeteer/blob/main/docs/api.md"; 
    let browser = await puppeteer.launch({
         headless: false,
         args: ['--start-maximized'],
         defaultViewport: {
            width: 1496,
            height: 766
        }
    });
    let page = await browser.newPage();
    await page.goto(url, { waitUntil: "networkidle0", timeout: 60000 });
    await page.screenshot({path: '1.png', fullPage: true})
    await page.close();
    await browser.close();
  ```
